client/state/data-getters/index.js

```diff
@@ -209,3 +209,62 @@ export const requestSiteAlerts = siteId => {
 		}
 	);
 };
+
+export const requestAtomicSFTPDetails = siteId =>
+	requestHttpData(
+		`atomic-hosting-data-${ siteId }`,
+		http(
+			{
+				method: 'GET',
+				path: `/sites/${ siteId }/hosting/ssh`,
+				apiNamespace: 'wpcom/v2',
+			},
+			{}
+		),
+		{
+			freshness: 5 * 60 * 1000,
+			fromApi: () => ( { username } ) => {
+				return [ `atomic-hosting-data-${ siteId }`, { username } ];
+			},
+		}
+	);
+
+export const resetAtomicSFTPUserPassword = siteId =>
+	requestHttpData(
+		`atomic-hosting-data-${ siteId }`,
+		http(
+			{
+				method: 'POST',
+				path: `/sites/${ siteId }/hosting/ssh/reset-password`,
+				apiNamespace: 'wpcom/v2',
+				body: {},
+			},
+			{}
+		),
+		{
+			fromApi: () => ( { username, password } ) => {
+				return [ `atomic-hosting-data-${ siteId }`, { username, password } ];
+			},
+			freshness: 0,
```

Comment:
you seemed to do a good job figuring it out already but I'd like to draw attention to the fact that this must be called as the result of an event and not included in `mapStateToProps` otherwise it'll `POST` the request repeatedly and do what we don't want.

good job!